src/librekey/rnp_key_store.cpp

```diff
@@ -717,6 +719,7 @@ rnp_key_store_remove_key(rnp_key_store_t *keyring, const pgp_key_t *key)
     if (!list_is_member(keyring->keys, (list_item *) key)) {
         return false;
     }
+    key->~pgp_key_t();
```

Comment:
I don't like the explicit destructor invocation. Is the plan to migrate the key store to use a vector instead and then get rid of this in the future?

Reply:
@dewyatt Yeah, me too. The plan is to update this in a separate PR, it would be too heavy to migrate all lists to std::vector in a single PR.